Src/IronPython/Compiler/Tokenizer.cs

```diff
@@ -963,8 +972,14 @@ internal Tokenizer(ErrorSink errorSink, PythonCompilerOptions options, bool verb
                         return new ConstantValueToken(LiteralParser.ParseImaginary(GetTokenString()));
 
                     default:
-                        BufferBack();
-                        MarkTokenEnd();
+                        if (!Char.IsDigit((char)ch) && !Char.IsWhiteSpace((char)ch) && ch != EOF) {
+                            BufferBack();
```

Comment:
I don't think this one is quite right. Just playing around with py3 there are other cases which are valid but fail `1 if 1e1else else 0`, `1 if 1e0else 0`, `1 if 1e10else 0`, etc.